cparser/processor.js

```diff
@@ -81,7 +81,7 @@ module.exports = that = {
 			var missingFuncs = regexParser.getMissingDeclarations(cleanText);
 
 			var prototypesStr = missingFuncs.join('\n') + '\n'
-				+ '#line ' + linesBeforeInjection + '\n';
+				+ '#line ' + linesBeforeInjection + ' "' + inputFile + '"\n';
```

Comment:
Yep it doesn't feel exactly right especially as the path is set in one buildpack but then compiled in another. It assumes that the same base is used so the same paths would be set but that might not be the case. But still, IMO it feels like less wrong than mixed paths mentioned above.